### PR TITLE
Auto save support

### PR DESCRIPTION
**Pull request by JimiVacarians**
Simple timer based auto save feature.

Can be modified in Option -> General.

### CODE REVIEW

File: FlaxEditor/Options/GeneralOptions.cs [hunk 1]
```diff
@@ -59,5 +59,13 @@ public sealed class GeneralOptions
         [DefaultValue(true)]
         [EditorDisplay("Analytics"), EditorOrder(1000), Tooltip("Enables or disables anonymous editor analytics service used to improve editor experience and the quality")]
         public bool EnableEditorAnalytics { get; set; } = true;
+        
+        [DefaultValue(true)]
+        [EditorDisplay("Auto Saves", "Enable Auto Saves"), EditorOrder(1000), Tooltip("Enables or disables auto saves")]
+        public bool EnabbleAutoSaves { get; set; } = true;
```
Comment (Erdroy) [inline FlaxEditor/Options/GeneralOptions.cs]: I would suggest adding comment to match the general style, same in the property below.

File: FlaxEditor/Options/GeneralOptions.cs [hunk 2]
```diff
@@ -65,7 +65,7 @@ public sealed class GeneralOptions
         public bool EnabbleAutoSaves { get; set; } = true;
         
         [DefaultValue(2)]
-        [EditorDisplay("Auto Saves", "Interval"), EditorOrder(1000), Tooltip("Set the interval between auto saves in minutes")]
+        [EditorDisplay("Auto Saves", "Interval in seconds"), EditorOrder(1000), Tooltip("Set the interval between auto saves in minutes")]
         public int AutoSavesInterval { get; set; } = 2;
```
Comment (Erdroy) [inline FlaxEditor/Options/GeneralOptions.cs]: There is also missing comment.

File: FlaxEditor/Options/GeneralOptions.cs (hunk 1 above)
Comment (mafiesto4) [reply FlaxEditor/Options/GeneralOptions.cs]: Typo in `EnabbleAutoSaves` (double `b`).

File: FlaxEditor/API/Editor.cs [hunk 3]
```diff
@@ -211,8 +214,45 @@ internal Editor()
...
+                _autoSaveTimer.Stop();
+                _autoSaveTimer.Dispose();
+                _autoSaveTimer = null;
+            }
+            else
+            {
+                if(_autoSaveTimer == null)
+                    InitAutoSaveTimer();
+
+                if (Math.Abs(_autoSaveTimer.Interval - Options.Options.General.AutoSavesInterval * 60 * 1000) > 0.01f) // floating point comparison
```
Comment (stefnotch) [inline FlaxEditor/API/Editor.cs]: `Mathf.NearEquals` or `Mathf.WithinEpsilon` could be used here.

File: FlaxEditor/Options/GeneralOptions.cs [hunk 4]
```diff
@@ -59,5 +59,19 @@ public sealed class GeneralOptions
...
+        /// </summary>
+        [DefaultValue(true)]
+        [EditorDisplay("Auto Saves", "Enable Auto Saves"), EditorOrder(1000), Tooltip("Enables or disables auto saves")]
+        public bool EnableAutoSaves { get; set; } = true;
+        
+        /// <summary>
+        /// Gets or sets a value indicating auto saves interval.
+        /// </summary>
+        [DefaultValue(2)]
+        [EditorDisplay("Auto Saves", "Interval In Seconds"), EditorOrder(1000), Tooltip("Set the interval between auto saves in seconds")]
```
Comment (stefnotch) [inline FlaxEditor/Options/GeneralOptions.cs]: Hold up, if that interval is supposed to be in seconds, then why does it get multiplied by 60 in a few other places?

File: FlaxEditor/API/Editor.cs [hunk 5]
```diff
@@ -211,8 +214,45 @@ internal Editor()
...
+                _autoSaveTimer.Dispose();
+                _autoSaveTimer = null;
+            }
+            else
+            {
+                if(_autoSaveTimer == null)
+                    InitAutoSaveTimer();
+
+                if (Math.Abs(_autoSaveTimer.Interval - Options.Options.General.AutoSavesInterval * 60 * 1000) > 0.01f) // floating point comparison 
+                    _autoSaveTimer.Interval = Options.Options.General.AutoSavesInterval * 60 * 1000;
```
Comment (stefnotch) [inline FlaxEditor/API/Editor.cs]: Why * 60? That would make sense if the AutoSavesInterval were in minutes.

File: FlaxEditor/API/Editor.cs [hunk 6]
```diff
@@ -211,8 +214,45 @@ internal Editor()
...
+            
+            InitAutoSaveTimer();
         }
 
+        private void InitAutoSaveTimer()
+        {
+            if (!Options.Options.General.EnableAutoSaves)
+                return;
+            
+            _autoSaveTimer = new Timer(Options.Options.General.AutoSavesInterval * 60 * 1000);
```
Comment (stefnotch) [inline FlaxEditor/API/Editor.cs]: Another instance of * 60 being used.

**Comment by mafiesto4** [comment]
Merged, thanks! I'll improve it and add info status update for user before auto-save.

**Pull request merged**